src/lib.rs

```diff
@@ -1316,6 +1275,32 @@ impl Build {
         }
     }
 
+    /// Run the compiler, generating intermediate files, but without linking
+    /// them into an archive file.
+    ///
+    /// This will return a list of compiled object files, in the same order
+    /// as they were passed in as `file`/`files` methods.
+    pub fn compile_intermediates(&self) -> Vec<PathBuf> {
+        match self.try_compile_intermediates() {
+            Ok(v) => v,
+            Err(e) => fail(&e.message),
+        }
+    }
+
+    /// Run the compiler, generating intermediate files, but without linking
+    /// them into an archive file.
+    ///
+    /// This will return a result instead of panicing; see `compile_intermediates()` for the complete description.
+    pub fn try_compile_intermediates(&self) -> Result<Vec<PathBuf>, Error> {
+        let dst = self.get_out_dir()?;
+        let objects = objects_from_files(self.files.iter().map(|v| &**v), &dst)?;
```

Comment:
I think it worth extracting `self.files.iter().map(|v| &**v)` into another function, or add another associated function `objects` which calculate this.

The `objects_from_file` is a generic function and since each closure has a different type, the function is mono-ed twice instead of once.

Reply:
I guess I could remove the genericity and just have object_from_files take an `&[Arc<Path>]`